docs/conflicts.rst

```diff
@@ -0,0 +1,131 @@
+The state database is also updated (conflicts can also be listed via the API and CLI).
+
+Although magic-folder itself doesn't try to examine the contents, you can now use any ``diff`` or similar tools you prefer to look at what is different between your copy and other participant(s) copies.
+
+
+Resolving Conflicts
+-------------------
+
+We cannot "magically" solve a conflict: two devices produced new Snapshots while not communicating with each other.
+
+Thus, it is up to the humans using these devices to determine what happens.
+Once appropriate changes are decided upon, a new Snapshot is produced with *two or more parents*: one parent for each of the Snapshots involved in the conflict (we've only talked about one other participant so far, but there could be more).
+
+Such a Snapshot (with two or more parents) indicates to the other clients a particular resoltion to the conflict has been decided.
+
+So there's actually another case when we see an incoming new Snapshot: it may in fact *resolve an existing* conflict.
+If this is the case, conflict markers are removed and the local database is updated (i.e. removing the conflict).
+
+It is a human problem if this resolution is not to your particular liking; you can produce an edit again or talk to the human who runs the other computer(s) involved.
+The history of these changes *is available* in the parent (or grandparent) Snapshots if the UX you're using can view or restore these.
+
+
+Resolving via Filesystem
+------------------------
+
+Not currently possible.
+
+
+Resolving via the CLI
+---------------------
+
+The subcommand ``magic-folder resolve`` may be used to specify a resolution.
+It allows you to choose ``--mine`` or ``--theirs`` (if there is only one other conflict).
+Otherwise, you must apply the ``--use <name>`` option to specify which version to keep.
+
+Currently there is no API for doing something more complex (e.g. simultaneuously replacing the latest version with new content).
+
+Complete example:
+
+.. code-block:: console
+
+    magic-folder resolve --mine ~/Documents/Magic/foo
+
+
+Resolving via the HTTP API
+--------------------------
+
+See :ref:`api_resolve_conflict`
+
+
+Future Directions
+-----------------
+
+We do not consider the current conflict functionality "done".
+There are other features required to make this more robust and have a nicer user experience.
+
+*Viewing old data*: While it is currently possible in the datamodel to view past versions of the files, we do not know of any UI that does this (and the CLI currently cannot).
+
+*Restore old version*: Similarly, it is possible to produce a new Snapshot that effectively restores an older version of the same file.
+We do not know of any UI that can do this.
+
+*Completely new content*: As hinted above, it might be nice to be able to produce a resolution that is some combination of multiple versions (like one sometimes does with Git conflicts, for example).
```

Comment:
In this context, I wonder also about pruning old snapshots (e.g., to save costs on metered/ZKAPs-enabled grids) and/or requesting deletion (e.g., in cases where a file might have been published unintentionally).

(These are somewhat of beyond just _conflict_ detection though...)

Reply:
There is no deletion API in Tahoe-LAFS.

The closest there is (which would work for "pruning" too) is to  simply stop refreshing leases on things you no longer care about.

Reply:
(I think this is more directly related to "history" / "how the datamodel works" than anything about conflicts per se though)

Reply:
>I think this is more directly related to "history" / "how the datamodel works" than anything about conflicts per se though

Yes, agreed (and hence my note/admission above that this is "beyond just conflict detection" -- so no need to really do anything here/now); this was mostly just an observation about something that I -- as a hypothetical/imaginary end user of this software -- might wonder about when encountering such a list of missing features. To that end, a note that there is no way to prune/delete snapshots _might_ be nice here (depending on what level of knowledge the reader is assumed to have) but I also grant that a) reliable "deletion" isn't (ever?) possible, b) if there ever were deletion-like functionality, it would probably be a Tahoe-LAFS thing -- not a Magic-Folder thing, and c) this is more of a general concern (that isn't specific to _conflicts_). So feel free to ignore/disregard/resolve. :)